jwt/jwt.go

```diff
@@ -227,43 +293,75 @@ func UserDetailsFromJWT(ctx context.Context) (firstName, lastName, email string,
 //        }
 //      ]
 //    },
-//    "issued_at": "2020-12-09T16:09:35.004042Z"
 //  },
-//  "sub": "695725d4-a345-4e68-919a-7395cb68484c"
-//}
-func ValidateToken(logger *logrus.Logger, t string) (*jwt.Token, error) {
-	keyForTokenWithLogger := func(token *jwt.Token) (interface{}, error) {
-		return keyForToken(logger, token)
-	}
-	token, err := jwt.Parse(t, keyForTokenWithLogger)
+// }
+func ValidateToken(logger *logrus.Logger, t string) (jwt.Token, error) {
+	_, err := accountsKeySet(logger)
```

Comment:
does this mean we are reading this file on every validation? Seems like something we might want to cache right?

Reply:
Oh I just realised that it is cached, I'm wondering though why we're not reading the key set on init. Something like that shouldn't be lazy loaded imo but rather prevent the server from coming up even. I guess this also goes hand in hand with my comment on the severity of the warning that we print when the file is not there. Shouldn't we simply exit if our accounts server doesn't have a keys file?

Reply:
Yes, that makes sense. While writing this I was thinking about the mixed case where we're running with both accounts and Kratos signing JWTs but it looks more and more likely that once we deliver this we will immediately axe Kratos. I will make it required.